liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/AbstractLibertySupport.java

```diff
@@ -209,6 +210,113 @@ protected Artifact getArtifact(String groupId, String artifactId, String type, S
+                findTransitiveDependencies(artifact, getProject().getArtifacts(), resolvedDependencyFiles);
+            } else {
+                log.warn("Unable to find artifact matching coordinates " + gavCoordinates + " in configured repositories.");
+            }
+        } else {
+            Set<Artifact> artifacts = getProject().getArtifacts();
+        
+            for (Artifact artifact : artifacts) {
+                if (artifact.getGroupId().equals(groupId) && 
+                    ((artifactId != null && artifact.getArtifactId().equals(artifactId)) ||
+                     (artifactId == null))) {
+                    if (!artifact.isResolved()) {
+                        ArtifactItem item = createArtifactItem(artifact.getGroupId(), artifact.getArtifactId(), artifact.getType(), artifact.getVersion()); 
+                        artifact = createArtifact(item);
+                    }
+                    log.debug("Found resolved dependency from project dependencies: " + artifact.getGroupId() + ":"
+                            + artifact.getArtifactId() + ":" + artifact.getVersion());
+                    resolvedDependencyFiles.add(artifact.getFile());
+                    findTransitiveDependencies(artifact, getProject().getArtifacts(), resolvedDependencyFiles);
+                }
+            }
+
+            if (resolvedDependencyFiles.isEmpty() && getProject().getDependencyManagement() != null) {
+                // if project has dependencyManagement section
+                List<Dependency> list = getProject().getDependencyManagement().getDependencies();
+            
+                for (Dependency dependency : list) {
+                    if (dependency.getGroupId().equals(groupId) && 
+                        ((artifactId != null && dependency.getArtifactId().equals(artifactId)) ||  
+                         (artifactId == null))) {
+                        ArtifactItem item = createArtifactItem(dependency.getGroupId(), dependency.getArtifactId(), dependency.getType(), dependency.getVersion()); 
+                        Artifact artifact = createArtifact(item);
+                        log.debug("Found resolved dependency from project dependencyManagement " + dependency.getGroupId() + ":"
+                            + dependency.getArtifactId() + ":" + dependency.getVersion());
+                        resolvedDependencyFiles.add(artifact.getFile());
+                        findTransitiveDependencies(artifact, getProject().getArtifacts(), resolvedDependencyFiles);
+                    }
+                }
+            }
+
+            if (resolvedDependencyFiles.isEmpty()) {
+                // No matching artifacts were found in the resolved dependencies. Send warning.
+                log.warn("Unable to find artifact matching coordinates " + gavCoordinates + " in either project dependencies or in project dependencyManagement.");
+            }
+        }
+
+        return resolvedDependencyFiles;
```

Comment:
So what happens if there are conflicts?   E.g.   A->B->D.1  and A->C->D.2  ?     Are we following the algorithm Maven uses to resolve conflicts (e.g. https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html ) or are we relying on the first/last one read?   (I'm just going to ask instead of trying to read).

I'm concerned too because when Maven resolves dependency conflicts, IIUC, if you read the typical resolution algorithm it resolves conflicts _for a single scope_.   If we're trying to collect dependencies across multiple scopes, what happens if you have say A->B->D.1  in compile scope, and C->E->D.2  in test scope?

Reply:
Honestly, I have no idea. This was the only way I could get any semblance of finding transitive dependencies for a particular dependency and am hoping that is good enough. The maven-dependency-plugin:copy-dependencies does not support copying transitive dependencies if either the include or exclude parameters are used to basically filter which dependencies to copy.

Reply:
Would the code copy both, or just the first (or last)?   If it copies both, can the filter be used to differentiate the one you want vs. the one you don't want?   I don't have a simple pre-existing example to test with or would mention it.    If we can't rationalize a behavior could we consider blowing up if we see two different versions somewhere?

Reply:
I like the idea of blowing up if we see two different versions of the same dependency, but I think that would potentially make this unusable for certain scenarios. I don't see how the filter can be used to differentiate which one you want. When I call getArtifacts() on the project, it is supposed to return all dependencies and transitive dependencies needed for the project. Would it have already weeded out the conflicts? I think we need a concrete scenario to try and see if this current code "does the right thing".

Reply:
Here's a simple pair of dependencies leading to a conflict:  https://maven.apache.org/enforcer/enforcer-rules/dependencyConvergence.html  

So two common mechanisms for selecting one or the other would be:

```
    <dependencyManagement>
       <dependencies>
           <dependency>
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>
             <version>1.6.1</version>
           </dependency>
        </dependencies>
    </dependencyManagement>
```

and

```
    <dependency>
      <groupId>org.slf4j</groupId>
      <artifactId>slf4j-nop</artifactId>
      <version>1.6.0</version>
      <exclusions>
        <exclusion>
            <groupId>org.slf4j</groupId>
            <artifactId>slf4j-api</artifactId>
        </exclusion>
    </exclusions>
    </dependency>
```
both selecting v 1.6.1.

It's a bit tricky since maven doesn't flag this situation as an error unless you add that dependencyConvergence enforcer rule... it'll pick one, I think the first?

So we might first stop and make sure that both the dependencyMgmt and exclusion approach work so that only one version of the dependency is seen, but if two are seen, IDK.. we could pick one.. is that the same as say `mvn dependency:tree` will show?  Not sure.

Reply:
Ok..I tried adding the following two dependencies to create the conflict...and only one version of slf4j-api was seen and copied.

```    <dependency>
      <groupId>org.slf4j</groupId>
      <artifactId>slf4j-nop</artifactId>
      <version>1.6.0</version>
    </dependency>
```
```    <dependency>
      <groupId>org.slf4j</groupId>
      <artifactId>slf4j-jdk14</artifactId>
      <version>1.6.1</version>
    </dependency>
```

Whichever one is configured first, that is the transitive dependency the code sees. When sfl4j-nop is first, we see version 1.6.0 of slf4j-api. When slf4j-jdk14 is first, we see version 1.6.1 of slf4j-api.

Then I tried adding the `dependencyManagement` section you listed, and it got version 1.6.1 of slf4j-api even when slf4j-nop was listed first. And finally I tried using the `exclusion` instead of `dependencyManagement`, and once again it got version 1.6.1 of slf4j-api. So all in all good news I think.